pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -31,6 +31,10 @@ const (
 	WorkerLabel = "node-role.kubernetes.io/worker"
 )
 
+type Interface interface {
+	Configure(types.WindowsVM) error
```

Comment:
I do create a mocked version, but in order to use the mocked version within the reconciler we need to pass in the nodeconfig entity in some way. As it was before the regular nodeconfig was being created within the reconciliation process. That doesn't give us any way to switch to using the mocked one. 

By making the nodeconfig part of the reconciler we can switch between using the regular and mocked version. Passing in the `types.WindowsVM` becomes necessary as we will now be using the same nodeconfig for multiple VMs, instead of a new one for every VM. We need to pass in the VM to select which VM will be configured.